package-config.cmake

```diff
@@ -70,21 +70,49 @@ MACRO(ADD_PROJECT_DEPENDENCY)
 
   # add dependency to the generated .pc
   # ref https://github.com/jrl-umi3218/jrl-cmakemodules/pull/335
-  cmake_parse_arguments(PARSED_ARGN "" "PKG_CONFIG_REQUIRES" "" ${ARGN})
+  # Add options OPTIONAL or REQUIRED
+  cmake_parse_arguments(PARSED_ARGN
+    "OPTIONAL;REQUIRED"
+    "PKG_CONFIG_REQUIRES"
+    ""
+    ${ARGN})
+
   IF(PARSED_ARGN_PKG_CONFIG_REQUIRES)
-    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_REQUIRES "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
-    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_DEP_NOT_FOR_CONFIG_CMAKE "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
+    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_REQUIRES
+      "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
+    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_DEP_NOT_FOR_CONFIG_CMAKE
+      "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
   ENDIF()
 
-  string(REPLACE ";" " " PACKAGE_ARGS "${PARSED_ARGN_UNPARSED_ARGUMENTS}")
-  list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_PACKAGE "  find_package(${PACKAGE_ARGS})")
-  list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_DEPENDENCY "  find_dependency(${PACKAGE_ARGS})")
-  find_package(${PARSED_ARGN_UNPARSED_ARGUMENTS})
+  string(REPLACE ";" " " PACKAGE_ARGS
+    "${PARSED_ARGN_UNPARSED_ARGUMENTS}")
+
+  # If REQUIRED is not specified consider the package as optional
+  # search for the package in a quiet mode.
+  IF(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
```

Comment:
Please avoid:
- using all caps IF/ELSE/ENDIF
- using the matching arguments in else and endif conditions, they have not been required for a really long time

Reply:
Thanks @gergondet for the feedback. I wanted to update my skills in cmake. 
For the sake of my understanding: 

* if the macro is using caps should we switch to lower case or stay coherent with the macros.

* What is the proper way to deal with arguments in a macro without if/else/endif conditions ? Could you point me towards an example ?

Reply:
> if the macro is using caps should we switch to lower case or stay coherent with the macros.

I think within a single macro we should stay coherent (or use the opportunity to switch all the macro to lower-cases)

Note: I realize it's not the case in this macro but I didn't review that part myself (and we could use some linter/format integration in the process)

> What is the proper way to deal with arguments in a macro without if/else/endif conditions ? Could you point me towards an example ?

In that case, I meant to replace:

```cmake
if(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
...
else(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
...
endif(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
```

with

```cmake
if(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
...
else()
...
endif()
```

the repeated conditions have not been necessary for a long time (CMake 2.6.0 see [this answer](https://stackoverflow.com/a/35269674))

Reply:
Thanks for the clarification !